packages/ui-kit/src/components/markdown.js

```diff
@@ -62,46 +51,57 @@ const H6 = styled.h6`
   font-weight: ${typography.fontWeights.regular};
   line-height: 1.4;
 `;
+
+/*
+The "container" styles have to be applied to containers that render markdown inside a surrounding
+visual box like a blockquote, notification box, card, or Subtitle.
+Heading margins are not set here because headings can and should not be used inside such containers.
+*/
+const containerStyles = () => css`
+  > * + * {
+    margin-top: ${dimensions.spacings.m};
+    margin-bottom: ${dimensions.spacings.m};
+  }
+  > :first-child {
+    margin-top: 0;
+  }
+  > :last-child {
+    margin-bottom: 0;
+  }
```

Comment:
If you use the lobotomized owl selector, you only need this:

```suggestion
  > * + * {
    margin-top: ${dimensions.spacings.m};
  }
```

Reply:
... except that it does not render the same in the smoke test site. 

One of the purposes of this wrapper is to _guarantee_ that the content does not have top or bottom margins, no matter what you throw in. 

But it's an indicator that the implementation is not consistent yet (e.g. the code example and  the headings still add margins by themselves).  I'll take a look whether it's realistic to cover all the cases in the components and then be able to not have to force-remove the very top and bottom margins.

Reply:
OK, here's my issue:

The code block component is imported from a different module (a.k.a. file).  If I want to remove the margin that's hardwired into the component and instead address it as a child in the `TypographyChild` CSS to set the margin there, emotion fails because it does not seem to be able to resolve components that are symbols, i.e. only those that are in the same file. 

No idea, it's a CSS-in-JS specific problem that I could resolve by starting to work with manual classes or by pulling the CodeBlock into the Markdown file.  All not nice.

Reply:
found it: https://github.com/styled-components/styled-components/issues/1326 

I'll address the emotion styled "Container" in the code block and not the CodeBlock itself